File: scripts/upload_images_to_lark.ts
import * as fs from 'fs';
import * as lark from "@larksuiteoapi/node-sdk";

const client = new lark.Client({
  appId: process.env.LARK_APPID || "",
  appSecret: process.env.LARK_SECRET || "",
  appType: lark.AppType.SelfBuild,
  domain: lark.Domain.Feishu,
});


// 获取启动时带的参数
const args = process.argv.slice(2);
if (args.length !== 1) {
  console.error('Usage: npm run upload_images_to_lark <directory>');
  process.exit(1);
}

// 读取目录下的所有文件
const directory = args[0];
fs.readdir(directory, (err, files) => {
  if (err) {
    console.error(`Unable to read directory: ${directory}`);
    process.exit(1);
  }

  // 过滤出所有图片文件
  const imageFiles = files.filter(file => {
    const extension = file.split('.').pop()?.toLowerCase();
    return extension === 'jpg' || extension === 'jpeg' || extension === 'png' || extension === 'gif';
  });

  // 输出所有图片文件的文件名
  console.log('Image files in directory:');
  [imageFiles[0]].forEach(file => {
    console.log(file);
    const buffer = fs.readFileSync(directory + "/" + file);
    //console.log(buffer);
    client.im.image.create({
      "data": {
        "image_type": "message",
        "image": buffer,
      }
    }).then(res => {
      console.log(res);
    }).catch(err => console.log("errorrrrrrrr", err))
  });
});

File: scripts/upload_images_to_lark.rb
require 'net/http'
require 'json'


dir_path = ARGV[0]  # 从命令行参数中获取目录路径

def get_access_token
  url = 'https://open.feishu.cn/open-apis/auth/v3/tenant_access_token/internal'
  uri = URI(url)
  http = Net::HTTP.new(uri.host, uri.port)
  http.use_ssl = true
  req = Net::HTTP::Post.new(uri.request_uri, 'Content-Type' => 'application/json')
  req.body = {
    app_id: ENV['LARK_APPID'],
    app_secret: ENV['LARK_SECRET']
  }.to_json
  res = http.request(req)
  JSON.parse(res.body)['tenant_access_token']
end

def upload_image(file)
  access_token = get_access_token

  #tempfile = Down.download(img_url)

  uri = URI('https://open.feishu.cn/open-apis/im/v1/images')
  http = Net::HTTP.new(uri.host, uri.port)
  http.use_ssl = true
  req = Net::HTTP::Post.new(uri, 'Content-Type' => 'multipart/form-data')
  req['Authorization'] = "Bearer #{access_token}"
  form_data = [
    ['image_type', 'message'],
    ['image', file]
  ]
  req.set_form(form_data, 'multipart/form-data')
  res = http.request(req)


  JSON.parse(res.body)['data']['image_key']
end


Dir.glob(File.join(dir_path, '**/*.{png,jpg,jpeg,gif,bmp}')) do |filename|
  puts(filename)
  f = File.read(filename)
  ret = upload_image(f)
  puts(ret)
end
